forward_script_python/forward_script/lib/query_builder.py

```diff
@@ -1 +1,35 @@
+# -*- coding: utf-8 -*-
+import urllib
 
+
+class QueryBuilder(object):
+
+    def get_query(self, info, page_name, object_id='', coordinates=None, additional_categories=None):
+        """
+        Generate a query string from available information
+        @type info: PageInformation
+        @type page_name: String
+        @type object_id: String
+        @type coordinates: dict
+        @type additional_categories: list
+        @return string url encoded querystring
+        """
+
+        if coordinates is None:
+            coordinates = {}
+        if additional_categories is None:
+            additional_categories = []
+
+        query = {}
+        categories = '|' . join([info.get_category()] + additional_categories)
+        query['categories'] = categories
+        if 'lat' in coordinates and coordinates['lat'] and 'lon' in coordinates and coordinates['lon']:
+            query['lat'] = coordinates['lat']
+            query['lon'] = coordinates['lon']
+        if info.has_usable_id():
```

Comment:
The `PageInformation` uses properties and not methods for the infos. This is different from PHP where everything is retuned by getters.
Please rewrite as `if info.has_usable_id:` etc

You'd then have to rewrite the tests from 
    `self._page_info.get_category.return_value = ''`
to
    `self._page_info.get_category = ''`